adminguide.rst

```diff
@@ -320,6 +322,52 @@ job queue and record of past jobs.
     and restart.
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+flux-accounting configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to support fairshare and multi-factor priority ordering of user jobs, the flux-accounting package must be present and configured for the system instance. This allows for the management of users and banks on a cluster, job priorities to be calculated using multiple factors and uses job record information from users to calculate and maintain fairshare values.
+
+The first thing to configure when first setting up the flux-accounting database is to set the PriorityUsageResetPeriod and PriorityDecayHalfLife parameters. Both of these parameters represent a number of weeks by which to hold usage factors up to the time period where jobs no longer play a factor in calculating a usage factor. If these parameters are not passed when creating the DB, PriorityDecayHalfLife is set to 1 week and PriorityUsageResetPeriod is set to 4 weeks, i.e the flux-accounting database will store up to a month's worth of jobs broken up into one week chunks:
+
+.. code-block:: console
+
+ $ flux account create-db --priority-decay-half-life=2 --priority-usage-reset-period=8 path/to/DB
+
+The other component to load is the multi-factor priority plugin, which can be loaded with ``flux jobtap load``:
+
+.. code-block:: console
+
+ $ flux jobtap load mf_priority.so
+
+This plugin can also be loaded by specifying it in the ``job-manager.plugins`` section of the Flux TOML configuration file. Example:
+
+.. code-block:: toml
+
+ [job-manager]
+ plugins = [
+   { load = "mf_priority.so"
+   },
+ ]
+
+After the DB and job priority plugin are set up, the ``update-usage`` subcommand should be set up to run as a cron job. This subcommand fetches the most recent job records for every user in the flux-accounting DB and calculates a new job usage value. This subcommand takes one optional argument, ``--priority-decay-half-life``, which, like the parameter set in the database creation step above, represents the number of weeks to hold one job usage "chunk." If not specified, this optional argument also defaults to 1 week.
+
+.. code-block:: console
+
+ $ flux account update-usage --priority-decay-half-life=2 path/to/DB
+
+After the job usage values are re-calculated and updated, the fairshare values for each user also need to be updated. This can be accomplished by configuring the ``flux-update-fshare`` script to also run as a cron job. This fetches user account data from the flux-accounting DB and recalculates and writes the updated fairshare values back to the DB.
+
+.. code-block:: console
+
+ $ flux account-update-fshare -f path/to/DB
+
+Once the fairshare values for all of the users in the flux-accounting DB get updated, this information will be sent to the priority plugin. The ``bulk_update.py`` script can also be configured to run as a cron job to send the updated values to the plugin:
+
+.. code-block:: console
+
+ $ flux account-priority-update -p path/to/DB
```

Comment:
We've got three different ways of specifying the path/to/DB in each script (positional, -f, and -p).  Is that a doc issue or a user interface issue?  Might want to fix that in the code if it really is that way.  Is there a default path to the database built into flux-accounting?  If the default is reasonable maybe we could just state up front what it is and leave off the options in the doc.

Reply:
That would be a user interface issue. Thanks for pointing this out. I will open an issue on flux-accounting to unify the path specification to the flux-accounting DB.

> Is there a default path to the database built into flux-accounting?

The default path to the database should be `/var/FluxAccounting.db`. Stephen had set that up when he added autotools support to the flux-accounting project.

Reply:
Maybe we should change it to `@X_LOCALSTATEDIR@/lib/flux/FluxAccounting.db` for your 0.11 release?  Then scripts would just work for the system instance packages without requiring options.

Reply:
If these commands should run be run as the flux user, then that should be specified as well.